### PR TITLE
Expand managed unit entry with User, Group + EnvironmentFile Array

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
I didn't see https://github.com/voxpupuli/puppet-systemd/pull/321 was already open for the user and group.

File: REFERENCE.md [hunk 1]
```diff
@@ -2201,7 +2203,10 @@ Struct[{
     Optional['OOMPolicy']                 => Enum['continue', 'stop','kill'],
     Optional['OOMScoreAdjust']            => Integer[-1000,1000],
     Optional['Environment']               => String,
-    Optional['EnvironmentFile']           => Variant[Stdlib::Unixpath,Pattern[/-\/.*/]],
+    Optional['EnvironmentFile']           => Variant[
+      Stdlib::Unixpath,Pattern[/-\/.+/],
+      Array[Variant[Stdlib::Unixpath,Pattern[/-\/.+/]],1],
```
Comment (kenyon) [inline REFERENCE.md:2208]: Does this allow for the special value `~`?

Comment (jhoblitt) [reply REFERENCE.md:2208]: I wasn't aware that `~` was used in `EnvironmentFile`. https://www.freedesktop.org/software/systemd/man/systemd.service.html doesn't mentioned it.

Comment (kenyon) [reply REFERENCE.md:2208]: Whoops, nevermind, got this confused with `WorkingDirectory` from #321.

**Review by kenyon — APPROVED** [review]

**Pull request merged**